cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -233,3 +246,12 @@ func SetDefaults_AuditPolicyConfiguration(obj *MasterConfiguration) {
 		obj.AuditPolicyConfiguration.LogMaxAge = &DefaultAuditPolicyLogMaxAge
 	}
 }
+
+// SetDefaults_HostPathMount sets default values to any extra volumes
+func SetDefaults_HostPathMounts(mounts []HostPathMount) {
+	for i := range mounts {
+		if mounts[i].ReadOnly == nil {
+			mounts[i].ReadOnly = pointer.BoolPtr(true)
```

Comment:
If omitted `ReadOnly` will be `false`, which will make the volume writable. This is not the current behavior, hence the piece of code.

Reply:
If I read this piece of code correctly: `mounts[i].ReadOnly == nil` will be true condition if ReadOnly is not explicitly set before, then line 254 will set it to `true`, making volume by default read-only. 
I feel that we need to get unit tests for those scenarios:
1. MasterConfig initialized with no *ExtraVolumes defined
2. MasterConfig initialized with some *ExtraVolumes defined, but not explicitly defining ReadOnly field
3. MasterConfig initialized with *ExtraVolumes and with explicitly set ReadOnly

Reply:
Yes, that is correct. This is the required behavior as previously `ReadOnly` was fixed to `true`.

I like the tests idea and will try to implement them as soon as possible (although I'm quite busy these days). I'll mark this as WIP until I have more time for it.

Reply:
My main reason for comments is that if person see in API description boolean variable, he/she might assume that if it is not specified, it will be treated as `false`. For *ExtraVolumes it is the opposite way, will be `true`, because of previously hardcoded value. Maybe adding comment to API field declaration can also be helpful.

Reply:
Exactly my thought. I was initially thinking if the option needs to be `writable`. However, the static pods already have a `readOnly` option. I beleive, that it defaults to `false` though.
Do you think, that it's best to rename the option to `writeable` and reverse its default altogether? This will remove the need for the boolean pointer too.